Additional cypress plugins

I am using three different cypress plugins `cypress-file-upload: 3.1.2`   `cypress-xpath: ^1.4.0 ``cypress-select-tests: ^1.3.1` Is it possible to add these to the image before the test run starts?

When running the following command: 
`docker run -it -v $PWD:/cypress -w /cypress --entrypoint="" cypress/included:3.4.1  cypress run --env PORTAL_PASSWORD=xxxxxx --browser chrome`

I get the following error: 

```
The plugins file is missing or invalid.

Your `pluginsFile` is set to `/monoweb/cypress/plugins/index.js`, but either the file is missing, it contains a syntax error, or threw an error when required. The `pluginsFile` must be a `.js` or `.coffee` file.

Please fix this, or set `pluginsFile` to `false` if a plugins file is not necessary for your project.

 Error: Cannot find module 'cypress-select-tests/grep'
    at Function.Module._resolveFilename (module.js:543:15)
    at Function.Module._load (module.js:473:25)
    at Module.require (module.js:586:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/monoweb/cypress/plugins/index.js:20:29)
    at Object.<anonymous> (/monoweb/cypress/plugins/index.js:26:3)
    at Module._compile (module.js:642:30)
    at Object.Module._extensions..js (module.js:653:10)
    at Module.load (module.js:561:32)
    at tryModuleLoad (module.js:504:12)
    at Function.Module._load (module.js:496:3)
    at Module.require (module.js:586:17)
    at require (internal/module.js:11:18)
    at module.exports (/root/.cache/Cypress/3.4.1/Cypress/resources/app/packages/server/lib/plugins/child/run_plugins.js:133:15)
    at Object.<anonymous> (/root/.cache/Cypress/3.4.1/Cypress/resources/app/packages/server/lib/plugins/child/index.js:9:25)
    at Object.<anonymous> (/root/.cache/Cypress/3.4.1/Cypress/resources/app/packages/server/lib/plugins/child/index.js:11:3)
```

--- a/.gitignore
+++ b/.gitignore
@@ -3,3 +3,4 @@
 test/src/.Trash-0/
 **/cypress/videos
 **/cypress/screenshots
+**/cypress/logs

--- a/README.md
+++ b/README.md
@@ -34,6 +34,8 @@ If you want to use `cypress/included` image, read [Run Cypress with a single Doc
 
 Folder [examples/run-included-as-non-root](examples/run-included-as-non-root) shows how to build a new Docker image on top of `cypress/included` image and run the tests as non-root user `node`.
 
+Folder [examples/included-with-plugins](examples/included-with-plugins) shows how to use locally installed [Cypress plugins](https://on.cypress.io/plugins) while running `cypress/included` image.
+
 ## Common problems
 
 ### Cannot run post-install hook

--- a/examples/included-with-plugins/README.md
+++ b/examples/included-with-plugins/README.md
@@ -1,2 +1,25 @@
 # Running cypress/included image with additional plugins
 
+You can run Cypress installed globally in `cypress/included:...` image and use locally installed plugins. You just need to install the plugins first. For example, in `src` folder we have End-to-end tests and [package.json](src/package.json) that lists [@cypress/skip-test](https://github.com/cypress-io/cypress-skip-test) and [cypress-failed-log](https://github.com/bahmutov/cypress-failed-log) development dependencies.
+
+First, install dependencies using NPM
+
+```shell
+cd src
+npm install
+```
+
+The local dependencies will be added to `node_modules` folder.
+
+Second, change back from `src` folder to the example folder and run tests
+
+```shell
+cd ..
+pwd
+docker run -it -v $PWD/src:/test -w /test cypress/included:3.8.0
+...
+```
+
+The tests should be able to load locally installed plugins from [cypress/plugins/index.js](src/cypress/plugins/index.js) and [cypress/support/index.js](src/cypress/support/index.js) files.
+
+**Tip:** if you need to install Cypress plugins locally, you might as well [install Cypress locally](https://on.cypress.io/installing-cypress). Just make sure to [cache it on CI](https://on.cypress.io/caching).

--- a/examples/included-with-plugins/README.md
+++ b/examples/included-with-plugins/README.md
@@ -1,0 +1,2 @@
+# Running cypress/included image with additional plugins
+

--- a/examples/included-with-plugins/src/.gitignore
+++ b/examples/included-with-plugins/src/.gitignore
@@ -1,0 +1,1 @@
+node_modules

--- a/examples/included-with-plugins/src/cypress.json
+++ b/examples/included-with-plugins/src/cypress.json
@@ -1,0 +1,3 @@
+{
+  "fixturesFolder": false
+}

--- a/examples/included-with-plugins/src/cypress/integration/spec.js
+++ b/examples/included-with-plugins/src/cypress/integration/spec.js
@@ -1,0 +1,19 @@
+import { onlyOn, skipOn } from '@cypress/skip-test'
+
+onlyOn('linux', () => {
+  describe('linux tests', () => {
+    it('has h2', () => {
+      cy.visit('index.html')
+      cy.contains('h2', 'test')
+    })
+  })
+})
+
+skipOn('linux', () => {
+  describe('Tests on Mac or Windows', () => {
+    it('has h2', () => {
+      cy.visit('index.html')
+      cy.contains('h2', 'test')
+    })
+  })
+})

--- a/examples/included-with-plugins/src/cypress/plugins/index.js
+++ b/examples/included-with-plugins/src/cypress/plugins/index.js
@@ -1,0 +1,6 @@
+console.log('in plugins file')
+module.exports = (on, config) => {
+  on('task', {
+    failed: require('cypress-failed-log/src/failed')(),
+  })
+}

--- a/examples/included-with-plugins/src/cypress/support/index.js
+++ b/examples/included-with-plugins/src/cypress/support/index.js
@@ -1,0 +1,1 @@
+require('cypress-failed-log')

--- a/examples/included-with-plugins/src/index.html
+++ b/examples/included-with-plugins/src/index.html
@@ -1,0 +1,4 @@
+<body>
+<h2>A test</h2>
+<p>This is a test page</p>
+</body>

--- a/examples/included-with-plugins/src/package.json
+++ b/examples/included-with-plugins/src/package.json
@@ -1,0 +1,16 @@
+{
+  "name": "src",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@cypress/skip-test": "2.3.0",
+    "cypress-failed-log": "2.5.1"
+  }
+}

--- a/examples/included-with-plugins/test.sh
+++ b/examples/included-with-plugins/test.sh
@@ -1,0 +1,9 @@
+set e+x
+
+echo "Running tests with additional locally installed Cypress plugins"
+echo "Note that plugins need to be installed first"
+echo "cd src; npm install"
+
+# print the docker command before running
+set -x
+docker run -it -v $PWD/src:/test -w /test cypress/included:3.8.0

--- a/examples/run-included-as-non-root/test.sh
+++ b/examples/run-included-as-non-root/test.sh
@@ -4,4 +4,6 @@ LOCAL_NAME=cypress/example
 
 echo "Running tests against $LOCAL_NAME"
 echo "Running as a non-root user 'node'"
+# print the docker command before running
+set -x
 docker run -it -v $PWD/src:/test -w /test -u node $LOCAL_NAME

